src/webapi.ts

```
@@ -649,6 +660,19 @@ export class Model {
      * Returns the matrix with the given name from the model.
      */
     async matrix(name: MatrixName): Promise<Matrix> {
+        // In the USEEIO model version >= 2.0, the matrix U is replaced with the N matrix
+        // So we check the model version, to decide wether we use the N or the U matrix
+        if (name === "N") {
+          try {
+            const modelVersion = parseInt(this._conf.model.split("v")[1]);
```

This is maybe a working fix for now but in general we cannot assume that the model ID always follows a pattern `[prefix]v[major].[minor]`, or can we? Better maybe just match with a regex `*v1.*` without throwing errors here?

---

You are right, I was assuming that the model ID had a predefined pattern.
However, I should throw an error if I can't find the version number, since it is used for knowing which matrix we should use (N or U), isn't it ?
Anyway, I am using the following pattern now `/v(\d*\.?\d*)/i`, to find the version number (it looks for a v followed by a number)

---

versioning will be like 
`v1.2`
`v2.0.1` 
so a wildcard is appropriate